spec/webauthn/fake_client_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe "FakeClient" do
+  let(:client) { WebAuthn::FakeClient.new }
+
+  context "#get" do
+    let!(:credential_1) { client.create }
+    let!(:credential_2) { client.create }
+
+    it "returns the first matching credential when allow_credentials is nil" do
+      assertion = client.get
+      expect(assertion["id"]).to eq(credential_1["id"])
+    end
+
+    it "returns the matching credential when allow_credentials is passed" do
+      allow_credentials = [{ id: credential_2["id"] }]
+      assertion = client.get(allow_credentials: allow_credentials)
```

Comment:
Thinking about the argument, I am wondering whether it needs to match so closely what it is in reality for `CredentialRequestOptions` or not...

Wondering whether just letting someone do `client.get(allow: credential_2["id"])` could be more simple providing the same feature?

- Do we really need to support multiple allowed credentials for the fake client?
- Do we really need to support the nested array and hash structure?

Keep in mind we already deviated from the real world when defining this testing-purpose FakeClient initially.

Reply:
> Do we really need to support multiple allowed credentials for the fake client?

I think that could be quite useful, yes and I don't think it's hurting us that much to support it.

> Do we really need to support the nested array and hash structure?

That's a fair point and I changed this to be only a array of ids, instead of the nested structure.

Reply:
Thanks!